src/compare-object-shallow/index.ts

```diff
@@ -18,6 +18,33 @@ import { ShallowEqualFn } from './types';
  * Input:
  * ({ a: { x: 1 } }, { a: obj }) => true // if obj is same reference as value of a
  */
+
 export const shallowEqual: ShallowEqualFn = (a, b) => {
-  throw new Error('Not Implemented');
+  if (a === b) {
+    return true;
+  }
+
+  if (
+    typeof a !== 'object' ||
+    a === null ||
+    typeof b !== 'object' ||
+    b === null
+  ) {
+    return false;
+  }
```

Comment:
Put it all together somehow or store it in readable variables. It's hard to read.

Reply:
Both have the type `Record\<string, unknown>` - it can't be anything other than an object.